cmd/snap/cmd_model.go

```diff
@@ -0,0 +1,228 @@
+			// make sure the header is in the map
+			if !ok {
+				continue
+			}
+
+			// switch on which header it is to handle some special cases
+			switch headerName {
+			// list of scalars
+			case "required-snaps":
+				headerIfaceList, ok := headerValue.([]interface{})
+				if !ok {
+					return fmt.Errorf("invalid type for \"%s\" header", headerName)
+				}
+				if len(headerIfaceList) == 0 {
+					fmt.Fprintf(w, "%s:\t[]\n", headerName)
+				} else {
+					fmt.Fprintf(w, "%s:\t\n", headerName)
+					for _, elem := range headerIfaceList {
+						headerStringElem, ok := elem.(string)
+						if !ok {
+							return fmt.Errorf("invalid type for \"%s\" header", headerName)
+						}
+						// note we don't wrap these, since for now this is
+						// specifically just required-snaps and so all of these
+						// will be snap names which are required to be short
+						fmt.Fprintf(w, "  - %s\n", headerStringElem)
+					}
+				}
+
+			//timestamp needs to be formatted with fmtTime from the timeMixin
+			case "timestamp":
+				timestamp, ok := allHeadersMap[headerName].(string)
+				if !ok {
+					return errors.New("invalid type for \"timestamp\" header")
+				}
+
+				// parse the time string as RFC3339, which is what the format is
+				// always in for assertions
+				t, err := time.Parse(time.RFC3339, timestamp)
+				if err != nil {
+					return err
+				}
+				fmt.Fprintf(w, "timestamp:\t%s\n", x.fmtTime(t))
+
+			// long string key we don't want to rewrap but can safely handle
+			// on "reasonable" width terminals
+			case "device-key-sha3-384":
+				// also flush the writer before continuing so the previous keys
+				// don't try to align with this key
+				w.Flush()
+				headerString, ok := headerValue.(string)
+				if !ok {
+					return errors.New("invalid type for \"device-key-sha3-384\" header")
+				}
+
+				switch {
+				case termWidth > 86:
+					fmt.Fprintf(w, "device-key-sha3-384: %s\n", headerString)
+				case termWidth <= 86 && termWidth > 66:
+					fmt.Fprintln(w, "device-key-sha3-384: |")
+					wrapLine(w, []rune(headerString), "  ", termWidth)
+				}
+
+			// long base64 key we can rewrap safely
+			case "device-key":
+				headerString, ok := headerValue.(string)
+				if !ok {
+					return errors.New("invalid type for \"device-key\" header")
+				}
+				// the string value here has newlines inserted as part of the
+				// raw assertion, but base64 doesn't care about whitespace, so
+				// split by newlines and re-wrap since base64 doesn't care about
+				// whitespace and we can make it prettier
+				headerString = strings.Join(strings.Split(headerString, "\n"), "")
+				fmt.Fprintln(w, "device-key: |")
+				wrapLine(w, []rune(headerString), "  ", termWidth)
+
+			// the default is all the rest of short scalar values, which all
+			// should be strings
+			default:
+				headerString, ok := headerValue.(string)
+				if !ok {
+					return fmt.Errorf("invalid type for \"%s\" header", headerName)
```

Comment:
`%q` rather than `\"%s\"` surely?

Reply:
ah yes you're right I always forget `%q` exists